types/coin.go

```diff
@@ -368,6 +368,23 @@ func (coins Coins) IsAllLTE(coinsB Coins) bool {
 	return coinsB.IsAllGTE(coins)
 }
 
+// IsAnyGT returns true iff for any denom in coins, the denom is present at a
+// greater amount in coinsB.
+func (coins Coins) IsAnyGT(coinsB Coins) bool {
+	if len(coinsB) == 0 {
+		return false
+	}
```

Comment:
This should be 

```golang
        // if len(coins) = 0 we know for sure no amount in coins can be GT coinsB
	if len(coins) == 0 {
		return false
	}

        // we learned from the above that coins contains some coins,
        // thus if len(coinsB) == 0 we know for sure there's something GT
	if len(coinsB) == 0 {
		return true
	}
```

Reply:
Again, it is important to be consistent across the methods. This behaves the same as `IsAnyGTE`. If `coinsB` then no denom is present and thus no coin is GT.

Reply:
This:

```golang
	if len(coins) == 0 {
		return false
	}
```

adds a bit of extra readability (thanks to early return) and does not conflict with `If coinsB then no denom is present and thus no coin is GT.`, in fact the `for _, coin := range coins ` would just not be executed. Also it does not break consistency with `IsAnyGTE`.

Though I see your point, this should be dropped:

```golang
	if len(coinsB) == 0 {
		return true
	}
```

Reply:
Ok, I'll update.

Reply:
Actually, 

```
if len(coins) == 0 {
		return false
	}
```

is redundant though.

Reply:
tbh I don't think this is necessary at all. The method is pretty concise as-is.